homeassistant/components/twitch/sensor.py

```diff
@@ -86,71 +90,76 @@ def setup_platform(
             _LOGGER.error("OAuth token is invalid")
             return
 
-    channels = client.get_users(logins=channels)
+    twitch_users: list[TwitchUser] = []
+    async for channel in client.get_users(logins=channels):
+        twitch_users.append(channel)
 
-    add_entities(
-        [TwitchSensor(channel, client) for channel in channels["data"]],
+    async_add_entities(
+        [TwitchSensor(channel, client) for channel in twitch_users],
         True,
     )
 
 
 class TwitchSensor(SensorEntity):
-    """Representation of an Twitch channel."""
+    """Representation of a Twitch channel."""
 
     _attr_icon = ICON
 
-    def __init__(self, channel: dict[str, str], client: Twitch) -> None:
+    def __init__(self, channel: TwitchUser, client: Twitch) -> None:
         """Initialize the sensor."""
         self._client = client
         self._enable_user_auth = client.has_required_auth(AuthType.USER, OAUTH_SCOPES)
-        self._attr_name = channel["display_name"]
-        self._attr_unique_id = channel["id"]
+        self._attr_name = channel.display_name
+        self._attr_unique_id = channel.id
 
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Update device state."""
-        followers = self._client.get_users_follows(to_id=self.unique_id)["total"]
-        channel = self._client.get_users(user_ids=[self.unique_id])["data"][0]
+        if self.unique_id is None:
+            return
```

Comment:
This is working around a typing issue. Maybe store `channel` (`TwitchUser`) as a local variable and use that directly instead?

Reply:
Oh, yes! Thanks, will do